src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,210 @@
+
+  const [shouldDisable, setShouldDisable] = useState<boolean>(noStickyColumns);
+  const [tableCellRefs, setTableCellRefs] = useState<Array<React.RefObject<HTMLTableCellElement>>>([]);
+  const { cellOffsets, updateCellOffsets } = useCellOffsets(tableCellRefs);
+
+  useResizeObserver(wrapperRef, () => updateCellOffsets());
+
+  const { first = 0, last = 0 } = stickyColumns || {};
+  // Calculate the indexes of the last left and right sticky columns, taking into account the selection column
+  const lastLeftStickyColumnIndex = first + (hasSelection ? 1 : 0);
+  const lastRightStickyColumnIndex = columnsLengthWithSelection - 1 - last;
+
+  // Get the width of the first and last sticky columns using the `cellOffsets` state, or use 0 if it's not available
+  const firstStickyColumnsWidth = cellOffsets.first[lastLeftStickyColumnIndex] ?? 0;
+  const lastStickyColumnsWidth = cellOffsets.last[lastRightStickyColumnIndex] ?? 0;
+
+  // Calculate the sum of all sticky columns' widths
+  const totalStickySpace = firstStickyColumnsWidth + lastStickyColumnsWidth;
+  useEffect(() => {
+    // Check if there is enough scrollable space for sticky columns to be enabled
+    const hasEnoughScrollableSpace =
+      totalStickySpace + MINIMUM_SCROLLABLE_SPACE + tablePadding.left + tablePadding.right < (containerWidth ?? 0);
+
+    // Determine if sticky columns should be disabled based on the conditions
+    const shouldDisable = noStickyColumns || !isWrapperScrollable || !hasEnoughScrollableSpace;
+    setShouldDisable(shouldDisable);
+  }, [containerWidth, noStickyColumns, stickyColumns, totalStickySpace, tablePadding, isWrapperScrollable]);
+
+  useEffect(() => {
+    // Create new refs for the visible columns and selection column, if present
+    setTableCellRefs(tableCellRefs =>
+      [...new Array(columnsLengthWithSelection)].map(
+        (_: any, i: number) => tableCellRefs[i] || createRef<HTMLTableCellElement>()
+      )
+    );
+  }, [columnsLengthWithSelection]);
+
+  const getStickyStyles = useCallback(
+    (colIndex: number, stickySide?: 'left' | 'right') => {
+      if (!stickySide) {
+        return { sticky: {}, stuck: {} };
+      }
+      const isFirstColumn = colIndex === 0;
+      let stuck = {};
+      if (isFirstColumn && !hasSelection && tablePadding.left !== 0) {
+        stuck = { paddingLeft: `${tablePadding.left}px` };
+      }
+      // Determine the offset of the sticky column using the `cellOffsets` state object
+      const offsetKey = stickySide === 'right' ? 'last' : 'first';
+      const stickyColumnOffset = cellOffsets[offsetKey]?.[colIndex + (hasSelection ? 1 : 0)];
+      return {
+        sticky: typeof stickyColumnOffset !== 'undefined' ? { [stickySide]: `${stickyColumnOffset}px` } : {},
+        stuck,
+      };
+    },
+    [cellOffsets, hasSelection, tablePadding]
+  );
+
+  const getStickyColumnProperties = useCallback(
+    (colIndex: number): GetStickyColumnProperties => {
+      const disabledStickyColumn = {
```

Comment:
Does this really need its own local variable? You could just move that to the `return` statement below.